napari/layers/base/base.py

```diff
@@ -252,7 +253,7 @@ def __init__(
         # Needs to be imported here to avoid circular import in _source
         from .._source import current_source
 
-        self.id = uuid.uuid4()
+        self.unique_id = self._get_unique_id()
```

Comment:
Maybe we should make this attribute private and add read_only property for it?

Reply:
@Czaki totally agree it should be read only, will update this. However, I'm not sure why we want to make it private? Based on the initial request in #5229 it seems like plugins may want to use it, and having it as a private attribute would discourage that I assume. Or are you suggesting this as a temporary measure, and we'd eventually make it public?

Reply:
Yes I think what we discussed is that this should be a property annotated with the type `collections.abc.hashable`